src/plugins/data_views/server/plugin.ts

```diff
@@ -46,7 +49,12 @@ export class DataViewsServerPlugin
     core.capabilities.registerProvider(capabilitiesProvider);
     const dataViewRestCounter = usageCollection?.createUsageCounter('dataViewsRestApi');
 
-    registerRoutes(core.http, core.getStartServices, dataViewRestCounter);
+    registerRoutes(
+      core.http,
+      core.getStartServices,
+      () => this.isRollupsEnabled(),
```

Comment:
wouldn't it be sufficient to directly use rollupsEnabled?
```suggestion
      () => this.rollupsEnabled,
```